rmgpy/data/kinetics/family.py

```diff
@@ -2228,7 +2236,458 @@ def getTrainingDepository(self):
                 return depository
         else:
             raise DatabaseError('Could not find training depository in family {0}.'.format(self.label))
+
+            
+    def addEntry(self,parent,grp,name):
+        """
+        Adds a group entry with parent parent
+        group structure grp
+        and group name name
+        """
+        ind = len(self.groups.entries)-1
+        entry = Entry(index=ind,label=name,item=grp,parent=parent)
+        self.groups.entries[name] = entry
+        self.rules.entries[name] = []
+        if entry.parent:
+            entry.parent.children.append(entry)
+    
+    def getEntriesReactions(self,template):
+        """
+        retrieves all training reactions whose kinetics
+        are associated with the entry template
+        """
+        entries = self.rules.entries[template]
+        tentries = self.getTrainingDepository().entries
+        rxnEntries = []
+        for entry in entries:
+            rxnEntries.append(tentries[entry.index].item)
+        return rxnEntries
+    
+    def getTemplateKinetics(self,template):
+        """
+        retrives a list of all the kinetics objects
+        associated with a given template
+        """
+        return [entry.data for entry in self.rules.entries[template]]
+    
+    def splitReactions(self,rxns,oldlabel,newgrp):
+        """
+        divides the reactions in rxns between the new
+        group structure newgrp and the old structure with 
+        label oldlabel
+        returns a list of reactions associated with the new group
+        the list of reactions associated with the old group
+        and a list of the indices of all of the reactions
+        associated with the new group
+        """
+        new = []
+        comp = []
+        newInds = []
+        kinetics = self.getTemplateKinetics(oldlabel)
+        
+        for i,rxn in enumerate(rxns):
+            if rxn.is_forward:
```

Comment:
You could reduce code duplication by doing the following:
```
reactants = rxn.reactants if rxn.is_forward else rxn.products
rmol = reactants[0].molecule[0]
for r in reactants[1:]:
    rmol.merge(r.molecule[0])
```

Reply:
done